ipam/actor.go

```diff
@@ -0,0 +1,165 @@
+package ipam
+
+import (
+	"fmt"
+	"github.com/zettio/weave/router"
+	"net"
+	"time"
+)
+
+// Start runs the allocator goroutine
+func (alloc *Allocator) Start() {
+	actionChan := make(chan func(), router.ChannelSize)
+	alloc.actionChan = actionChan
+	go alloc.actorLoop(actionChan)
+}
+
+// Actor client API
+
+// GetFor (Sync) - get IP address for container with given name
+func (alloc *Allocator) GetFor(ident string, cancelChan <-chan bool) net.IP {
```

Comment:
This chan comes from go's http package's closeNotifier (http://golang.org/src/net/http/server.go?s=3493:3661#L96) - is it worth the effort translating this to a chan struct{}?